mig-agent/config.go

```diff
@@ -167,9 +183,30 @@ type globals struct {
 
 	// Add the private client key below.
 	agentKey []byte
+
+	// let you target specific agents
+	tags map[string]string
+}
+
+func globalTagsMap() map[string]string {
+	tags := make(map[string]string)
+	inspectTags := reflect.TypeOf(&TAGS).Elem()
+
+	for tag, val := range structs.Map(TAGS) {
+		field, ok := inspectTags.FieldByName(tag)
+
+		if ok {
+			_, tag = stringPair(string(field.Tag))
+			tag = strings.Replace(tag, `"`, "", -1)
+		}
+		tags[tag] = fmt.Sprintf("%v", val)
+	}
+	return tags
```

Comment:
I'm wondering if there is a way we could do this without needing to use reflect and structs?

Like for example, maybe we just say TAGS has to include a Runtime element that is defined as a slice of key/value strings. Then when the agent starts up, we just add anything in the config file into that slice which we will know exists already.

TAGS right now is just an arbitrary struct, but historically we've never really used it much for anything other than key/value pairs. Maybe we should just change the TAGS type to be a map an avoid the conversion, and it would make it easier to work with?